nomad/client_alloc_endpoint_test.go

```diff
@@ -196,7 +197,19 @@ func TestClientAllocations_GarbageCollectAll_Remote(t *testing.T) {
 
 	testutil.WaitForResult(func() (bool, error) {
```

Comment:
This is very similar to https://godoc.org/github.com/hashicorp/nomad/e2e/e2eutil#WaitForNodesReady  - can we use it here?

Reply:
We can't because that uses the `api.Client` which uses the HTTP API.

Most -- if not all -- `nomad/` and `client/` tests only expose the RPC API as the HTTP API is setup by the Agent and introduces a number of extra dependencies to provide/configure.

Probably worth factoring this out into a helper as it may fix other tests.